utils_cv/detection/plot.py

```diff
@@ -0,0 +1,279 @@
+) -> JpegImageFile:
+    """ Plot boxes on Image and return the Image
+
+    Args:
+        im: The image to plot boxes on
+        boxes: the boxes to plot - a list of [xmin, ymin, xmax, ymax] boxes
+        categories: list of catogires to label the boxes with
+        bbox_params: the parameter of the bounding boxes
+
+    Returns:
+        The same image with boxes and labels plotted on it
+    """
+    # Convert to RGB
+    im = cv2.cvtColor(im, cv2.COLOR_BGR2RGB)
+    if len(boxes) > 0:
+        for box, category in zip(boxes, categories):
+
+            # reformat boxes to be consumable by cv2
+            box = [(box[0], box[1]), (box[2], box[3])]
+
+            # Draw Rectangle with the coordinates
+            cv2.rectangle(
+                im,
+                box[0],
+                box[1],
+                color=bbox_params.rect_color,
+                thickness=bbox_params.rect_th,
+            )
+
+            # Write the prediction class
+            cv2.putText(
+                im,
+                category,
+                (box[0][0], box[0][1] + 15),
+                cv2.FONT_HERSHEY_SIMPLEX,
+                bbox_params.text_size,
+                color=bbox_params.text_color,
+                thickness=bbox_params.text_th,
+            )
+
+    return im
+
+
+def display_bounding_boxes(
+    boxes: List[List[int]],
+    categories: List[str],
+    im_path: Union[Path, str],
+    ax: Union[None, plt.axes] = None,
+    bbox_params: BoundingBoxParams = BoundingBoxParams(),
+    figsize: Tuple[int, int] = (12, 12),
+) -> None:
+    """ Draw image with bounding boxes.
+
+    Args:
+        boxes: A list of [xmin, ymin, xmax, ymax] bounding boxes to draw
+        categories: A list of detected categories
+        im_path: the location of image path to draw
+        ax: an optional ax to specify where you wish the figure to be drawn on
+
+    Returns nothing, but plots the image with bounding boxes and categories.
+    """
+    # Read image with cv2
+    im = cv2.imread(str(im_path))
+
+    # plot boxes on im
+    im = plot_boxes(im, boxes, categories, bbox_params=bbox_params)
+
+    # display the output image
+    if ax is not None:
+        ax.set_xticks([])
+        ax.set_yticks([])
+        ax.imshow(im)
+    else:
+        plt.figure(figsize=figsize)
+        plt.imshow(im)
+        plt.xticks([])
+        plt.yticks([])
+        plt.show()
+
+
+def _get_cmap(n, name="hsv"):
+    """Returns a function that maps each index in 0, 1, ..., n-1 to a distinct
+    RGB color; the keyword argument name must be a standard mpl colormap name."""
+    return plt.cm.get_cmap(name, n)
+
+
+def _setup_pr_axes(ax: plt.axes, title: str) -> plt.axes:
+    """ Setup the plot settings for plotting PR curves. """
+    ax.set_xlabel("recall", fontsize=12)
+    ax.set_ylabel("precision", fontsize=12)
+    ax.set_xlim(0, 1.0)
+    ax.set_ylim(0, 1.01)
+    ax.set_title(title, fontsize=14)
+    ax.grid(True)
+    return ax
+
```

Comment:
Could add logical separator here that everything below is for Precision-Recall curve plotting. E.g.

# ===== Precision - Recall curve =====